src/main.js

```diff
@@ -1,84 +1,73 @@
-// Private Methods
-function extendConfig(defaults, options) {
```

Comment:
Having a separate private method for this is not necessary. It can be reduced to a single for loop within the constructor. (less memory use, less computation, less code to read)

Also, guarding against inherited prototypal props within for-in loops results in extremely messy code.
The JSLint rules I had to follow in Cognifide's repo enforced this pattern. It drove me absolutely nuts.

There are very few scenarios where this would actually pose a problem so I think we can safely live without it.